components/mdns/mdns.c

```diff
@@ -17,6 +17,7 @@
 #include "mdns_networking.h"
 #include "esp_log.h"
 #include "esp_random.h"
+#include "lwip/opt.h"
```

Comment:
Please don't use lwip headers from this file, it's supposed to be network stack independent.

You can use `CONIG_LWIP_IPV4` instead or define a `MNDS_IPV4` based on the config option in `mdns_config.h`

Reply:
`CONFIG_LWIP_IPV4` was introduced in IDF v5.1 but this managed component is for `IDF version>=5.0`, so I didn't use this config option here. May I create a menuconfig option in this component instead of using `CONIG_LWIP_IPV4`?